pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -123,6 +124,15 @@ func (c *controller) ValidateMachineClass(classSpec *v1alpha1.ClassSpec) (*v1alp
 		return nil, nil, retry, err
 	}
 
+	clone := machineClass.DeepCopy()
+
+	if finalizers := sets.NewString(clone.Finalizers...); !finalizers.Has(MCMFinalizerName) {
+		klog.Errorf("The machine class %s has no finalizers set. So not reconciling the machine.", machineClass.Name)
+		c.machineClassQueue.Add(machineClass.Name)
+		err := errors.New("The machine class %s has no finalizers set. So not reconciling the machine." + machineClass.Name)
+		return nil, nil, retry, err
+	}
+
```

Comment:
```suggestion
	if finalizers := sets.NewString(machineClass.Finalizers...); !finalizers.Has(MCMFinalizerName) {
		klog.Errorf("The machine class %s has no finalizers set. So not reconciling the machine.", machineClass.Name)
		err := errors.New("The machine class %s has no finalizers set. So not reconciling the machine." + machineClass.Name)
		c.machineClassQueue.Add(machineClass.Name)
		return nil, nil, retry, err
	}

```
Do we need to clone this machineClass object?

Reply:
Changes made in the latest commit.